[REG-2061] When creating a bot sequence from the Latest Recording sort the segments numerically.

batu opened the pull request:
[Loom Video](https://www.loom.com/share/1fe8e487f74e4a78abf47a8cc374563f?sid=dd98d33b-a5b0-4f45-b952-76110476a312)

Currently when we are creating a BotSequence from the copied bot segments the segments are sorted alphanumerically. This means that 2.json will come after 19.json, breaking the sequence.

This PR hotfixes it by ordering it numerically. 

We should do the same sorting order wherever we are showing segments.

RG-nAmKcAz commented on src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs:
```diff
@@ -278,7 +278,12 @@ await CreateAndUploadGameplaySession(
             // get all the file paths normalized to /
-            var segmentFiles = Directory.EnumerateFiles(botSegmentsDirectoryPrefix).Where(a=>a.EndsWith(".json")).Select(a=>a.Replace('\\','/')).Select(a=>a.Substring(a.LastIndexOf('/')+1));
+            var segmentFiles = Directory.EnumerateFiles(botSegmentsDirectoryPrefix)
+                .Where(a=>a.EndsWith(".json"))
```
Use a call to `BotSegmentDirectoryParser.OrderJsonFiles` instead of this new code block

batu commented:
Changed to using `BotSegmentDirectoryParser`.

batu commented:
@RG-nAmKcAz @vontell can you please take another look? Some other change affected this PR, and in order to get it to work I had to add `.ToList()` right after sorting.

Here is a [loom video](https://www.loom.com/share/f5e784880f774757a59e9816dcd749d0) showing the behavior.

batu commented:
I don't want to merge this before understanding what is going on, but my testing shows that it works as intended now. So feel free to merge if you think this looks good

RG-nAmKcAz commented on src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ScreenRecorder.cs:
```diff
@@ -288,8 +289,14 @@ await CreateAndUploadGameplaySession(
+                .Select(a=>a.Substring(a.LastIndexOf('/')+1));
+
+            // Order numerically instead of alphanumerically to ensure 2.json is before 10.json.
+            segmentFiles = BotSegmentDirectoryParser.OrderJsonFiles(segmentFiles).ToList();
```
calling .ToList() would make no difference in ordering and adds a layer of inefficiency, marking as requested changes until there is an explanation on why it wasn't working without this

vontell replied:
I just spent a good few minutes looking at this - to me it smelled like something to do with the fact that these LINQ functions defer execution until the elements are actually needed, but I couldn't find anything glaring in the code (for instance, we actually force the execution of the LINQ functions inside of OrderJsonFiles because of some print statements that print the elements coming in, so they are already being forced).

From the video, the strange part is that the logs just stop.... I would double check if an exception is being thrown somehow and if it's being suppressed by some surrounding try catch.

RG-nAmKcAz replied:
Linq functions are also notorious for eating exceptions

batu commented:
@RG-nAmKcAz PTAL. We understand the problem and the fix now. 

Because of lazy evaluation we would evaluate `Directory.EnumerateFiles(botSegmentsDirectoryPrefix)` AFTER the move operation, which would invalidate the path. 

Changed `Directory.EnumerateFiles` to `Directory.GetFiles` for instant execution of loading files. Added a comment
